apps/src/lib/levelbuilder/lesson-editor/AddResourceDialog.jsx

```diff
@@ -18,33 +60,212 @@ const styles = {
 export default class AddResourceDialog extends Component {
   static propTypes = {
     isOpen: PropTypes.bool.isRequired,
-    handleConfirm: PropTypes.func.isRequired
+    onSave: PropTypes.func,
+    handleClose: PropTypes.func.isRequired,
+    typeOptions: PropTypes.arrayOf(PropTypes.string),
+    audienceOptions: PropTypes.arrayOf(PropTypes.string)
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      key: '',
+      name: '',
+      type: '',
+      audience: '',
+      pdf: false,
+      assessment: false,
+      url: '',
+      downloadUrl: '',
+      error: ''
+    };
+  }
+
+  validateResource = () => {
+    const {key, name, url} = this.state;
+    let error = '';
+    if (name === '') {
+      error += 'Name is required. ';
+    }
+    if (url === '') {
+      error += 'URL is required. ';
+    }
+    if (key === '') {
+      error += 'Embed slug is required.';
+    }
+    if (error === '') {
+      return true;
+    } else {
+      this.setState({error});
+      return false;
+    }
+  };
+
+  resetState = () => {
+    this.setState({
+      key: '',
+      name: '',
+      type: '',
+      audience: '',
+      pdf: false,
+      assessment: false,
+      url: '',
+      downloadUrl: '',
+      error: ''
+    });
+  };
+
+  saveResource = e => {
+    e.preventDefault();
```

Comment:
is this needed to prevent the submit button click from navigating away?

Reply:
Yeah the page reloads if I don't add this